projects/wave-core-new/src/lib/datasets/upload/upload.component.ts

```diff
@@ -0,0 +1,188 @@
+import {HttpEventType} from '@angular/common/http';
+import {Component, OnInit, ChangeDetectionStrategy} from '@angular/core';
+import {Subject} from 'rxjs';
+import {DataSetIdDict, UUID} from '../../backend/backend.model';
+import {NotificationService} from '../../notification.service';
+import {DataSetService} from '../dataset.service';
+
+interface ExampleLoadingInfo {
+    name: string;
+    json: string;
+}
+
+@Component({
+    selector: 'wave-upload',
+    templateUrl: './upload.component.html',
+    styleUrls: ['./upload.component.scss'],
+    changeDetection: ChangeDetectionStrategy.OnPush,
+})
+export class UploadComponent implements OnInit {
+    selectedFiles: FileList;
+    progress$ = new Subject();
```

Comment:
what is the data type?

You don't need to annotate
`progress$: Subject<Datatype> = new Subject();`
just
`progress$ = new Subject<Datatype>();`
suffices

Reply:
7b114b945855735e9b45be3e93d131ce8ee7199c